src/components/AuthRoute.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useState, useEffect } from "react";
+import { getAuth, onAuthStateChanged } from "firebase/auth";
+import { useNavigate } from "react-router-dom";
+export interface IAuthRouteProps { }
+
+
+
+const AuthRoute: React.FunctionComponent<IAuthRouteProps> = (props) => {
+	const { children } = props;
```

Comment:
**suggestion (non-blocking):** Thanks to the up-front typing of the component, this could be deconstructed in the line above:

```
const AuthRoute: React.FunctionComponent<IAuthRouteProps> = ({ children }) => {
```